utils/UUID.hh

```diff
@@ -53,20 +53,43 @@ public:
         return version() == 1;
     }
 
+    bool is_timestamp_v7() const noexcept {
+        return version() == 7;
+    }
+
     bool is_timestamp() const noexcept {
-        return is_timestamp_v1();
+        return is_timestamp_v1() || is_timestamp_v7();
```

Comment:
Need to audit the callers of is_timestamp() to see whether they expect any timestamp or timestamp v1. For example, CQL timeuuid probably wants only v1.